cape/cape_result.py

```diff
@@ -2260,7 +2310,58 @@ def process_buffers(
                     ):
                         buffer_body.append(table_row)
                         count_per_source_per_process += 1
+                        b_buffer = bytes(buffer, 'utf-8')
+                        if all(PE_indicator in b_buffer for PE_indicator in PE_INDICATORS):
+                            hash = sha256(b_buffer).hexdigest()
+                            network_buffers.append((f"{str(process)}-{api_call}-{hash}", b_buffer, buffer))
+
+    if not os.path.exists(BUFFER_PATH):
+        os.mkdir(BUFFER_PATH)
+
+    for filename, b_buffer, buffer in network_buffers:
+        pebuffer = bytearray(b_buffer)
+        try:
+            PE_from_buffer = pefile.PE(data=pebuffer)
+            if is_valid(PE_from_buffer):
+                PE_from_buffer.write(f"{BUFFER_PATH}/{filename}")
+        except Exception as E:
+            try:
+                if lief.is_pe(pebuffer):
+                    PE_from_buffer = lief.PE.parse(pebuffer)
+                    PE_from_buffer.build()
+                    PE_from_buffer.write(f"{BUFFER_PATH}/{filename}")
+                elif lief.is_pe(buffer):
+                    PE_from_buffer = lief.PE.parse(buffer)
+                    PE_from_buffer.build()
+                    PE_from_buffer.write(f"{BUFFER_PATH}/{filename}")
+                else:
+                    with open(f"{BUFFER_PATH}/{filename}", "wb+")as f:
+                        f.write(buffer)
+            except Exception as E:
+                continue
+    for filename, b_buffer, buffer in buffers:
+        pebuffer = bytearray(b_buffer)
```

Comment:
looks like this entire buffer-handling code is duplicated between network_buffers and buffers. Can you please modularize this into a method and just call it for both instances?